cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/connector/resource/AzureDatabaseResourceService.java

```diff
@@ -150,24 +156,34 @@
                 .build(), ResourceStatus.DELETED));
     }
 
-    private List<CloudResourceStatus> deleteDatabaseServer(List<CloudResource> resources, CloudContext cloudContext, boolean force,
+    private List<CloudResourceStatus> deleteResources(List<CloudResource> resources, CloudContext cloudContext, boolean force,
             AzureClient client, PersistenceNotifier persistenceNotifier) {
-        LOGGER.debug("Deleting database server");
-        Optional<CloudResource> dbServerResourceOptional = resources.stream().filter(r -> AZURE_DATABASE.equals(r.getType())).findFirst();
-        if (dbServerResourceOptional.isEmpty()) {
-            LOGGER.warn("Azure database id not found in database, deleting nothing");
-            return List.of();
-        }
-        String databaseServerId = dbServerResourceOptional.get().getReference();
-        Optional<String> errorMessage = azureUtils.deleteDatabaseServer(client, databaseServerId, force);
-        if (errorMessage.isEmpty()) {
-            deleteResources(resources, cloudContext, persistenceNotifier);
-        }
 
-        return Lists.newArrayList(new CloudResourceStatus(CloudResource.builder()
-                .type(AZURE_DATABASE)
-                .name(databaseServerId)
-                .build(), ResourceStatus.DELETED));
+        // TODO simplify after final form of template is reached
+
+        List<CloudResourceStatus> deletedResourcesList = new ArrayList<>();
+
+        List<CloudResource> azureGenericResources = findResources(resources, List.of(AZURE_PRIVATE_ENDPOINT));
+        LOGGER.debug("Deleting dns zone groups and azure private endpoints {}", azureGenericResources);
+        azureUtils.deleteGenericResources(client, azureGenericResources.stream().map(CloudResource::getReference).collect(Collectors.toList()));
+        azureGenericResources.forEach(cr -> persistenceNotifier.notifyDeletion(cr, cloudContext));
+
+        findResources(resources, List.of(AZURE_DATABASE))
+                .forEach(r -> {
+                    LOGGER.debug("Deleting postgres server {}", r.getReference());
+                    azureUtils.deleteDatabaseServer(client, r.getReference(), force);
+                    persistenceNotifier.notifyDeletion(r, cloudContext);
+                    deletedResourcesList.add(new CloudResourceStatus(CloudResource.builder()
+                            .type(AZURE_DATABASE)
+                            .name(r.getReference())
+                            .build(), ResourceStatus.DELETED));
+                });
```

Comment:
I prefer stream+collect rather than foreach and adding element to an external list
eg (please note I'm using default naming generated by idea here):
```
        List<CloudResourceStatus> collect = findResources(resources, List.of(AZURE_DATABASE))
                .stream()
                .map(r -> getCloudResourceStatus(cloudContext, force, client, persistenceNotifier, r))
                .collect(Collectors.toList());

        return collect;
    }

    private CloudResourceStatus getCloudResourceStatus(CloudContext cloudContext, boolean force, AzureClient client, PersistenceNotifier persistenceNotifier, CloudResource r) {
        LOGGER.debug("Deleting postgres server {}", r.getReference());
        azureUtils.deleteDatabaseServer(client, r.getReference(), force);
        persistenceNotifier.notifyDeletion(r, cloudContext);
        return new CloudResourceStatus(CloudResource.builder()
                .type(AZURE_DATABASE)
                .name(r.getReference())
                .build(), ResourceStatus.DELETED);
    }